src/coreclr/jit/assertionprop.cpp

```diff
@@ -4699,6 +4699,7 @@ GenTree* Compiler::optAssertionProp_Update(GenTree* newTree, GenTree* tree, Stat
             if (parent != nullptr)
             {
                 parent->ReplaceOperand(useEdge, newTree);
+                parent->gtFlags |= gtGetFlagsForOperand(parent->gtOper, newTree);
```

Comment:
This flag updating call should be moved to this place in morph:

https://github.com/dotnet/runtime/blob/7980417f4c39c730617059ac983e0bcb4d20f419/src/coreclr/jit/morph.cpp#L9568-L9571